short_url/url/models.py

```diff
@@ -1,9 +1,11 @@
 from django.db import models
 
+from hashids import Hashids
+
 
 class Url(models.Model):
-    url = models.CharField(max_length=512)
-    short_url = models.CharField(max_length=64, blank=True)
+    url = models.URLField(max_length=512, unique=True)
+    short_url = models.URLField(max_length=64, blank=True, unique=True)
```

Comment:
I don't think we still need to store `short_url` anymore, since we can always use the ID to generate it. Hashids operations are fast enough to use it everytime we want to display a `short_url` representation to a user.

What I would recommend doing, would be to remove this field, and add a `@property` to this class.

Giving that you moved the `hashids` instance to somewhere else (there's another comment regarding it):

```python
from utils import hashids


class Url(models.Model)
    ...
    
    @property
    def short_url(self):
        return hashids.encode(self.pk)
```

`@property` in python makes so that you can access a method as it where a variable, meaning you will be able to access `short_url` using `url.short_url`.